packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -522,7 +536,11 @@ If you provided a React element for the optionText prop, you must also provide t
             if (inputValue === '' && filterValue === '' && createLabel) {
                 // create option with createLabel
                 filteredOptions = filteredOptions.concat(getCreateItem(''));
-            } else if (!doesQueryMatchSuggestion(filterValue)) {
+            } else if (
+                inputValue &&
+                filterValue &&
+                !doesQueryMatchSuggestion(filterValue)
+            ) {
```

Comment:
This is actually not directly linked to the main fix, but fixes an issue where `<AutocompleteArrayInput>` would still render the 'Create' option when the filter is empty (i.e. the `createLabel` option), although this should no longer be the case according to this [PR](https://github.com/marmelab/react-admin/pull/10266).